plugins/modules/gitlab_project_badge.py

```diff
@@ -0,0 +1,216 @@
+    api_token: 'Your-Private-Token'
+    project: projectname
+    state: absent
+    link_url: 'https://example.gitlab.com/%{project_path}'
+    image_url: 'https://example.gitlab.com/%{project_path}/badges/%{default_branch}/pipeline.svg'
+'''
+
+RETURN = '''
+badge:
+  description: The badge information.
+  returned: when I(state=present)
+  type: dict
+  sample:
+    id: 1
+    link_url: 'http://example.com/ci_status.svg?project=%{project_path}&ref=%{default_branch}'
+    image_url: 'https://shields.io/my/badge'
+    rendered_link_url: 'http://example.com/ci_status.svg?project=example-org/example-project&ref=master'
+    rendered_image_url: 'https://shields.io/my/badge'
+    kind: project
+'''
+
+from ansible.module_utils.api import basic_auth_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible_collections.community.general.plugins.module_utils.gitlab import (
+    auth_argument_spec, gitlab_authentication, find_project, ensure_gitlab_package
+)
+
+
+def present_strategy(module, gl, project, wished_badge):
+    changed = False
+
+    existing_badge = None
+    for badge in project.badges.list(iterator=True):
+        if badge.image_url == wished_badge["image_url"]:
+            existing_badge = badge
+            break
+
+    if not existing_badge:
+        changed = True
+        if module.check_mode:
+            return changed, {"status": "A project badge would be created."}
+
+        badge = project.badges.create(wished_badge)
+        return changed, badge.attributes
+
+    if existing_badge.link_url != wished_badge["link_url"]:
+        changed = True
+        existing_badge.link_url = wished_badge["link_url"]
+
+    if changed:
+        if module.check_mode:
+            return changed, {"status": "Project badge attributes would be changed."}
+
+        existing_badge.save()
+
+    return changed, existing_badge.attributes
+
+
+def absent_strategy(module, gl, project, wished_badge):
+    changed = False
+
+    existing_badge = None
+    for badge in project.badges.list(iterator=True):
+        if badge.image_url == wished_badge["image_url"]:
+            existing_badge = badge
+            break
+
+    if not existing_badge:
+        return changed, None
+
+    changed = True
+    if module.check_mode:
+        return changed, {"status": "Project badge would be destroyed."}
+
+    existing_badge.delete()
+
+    return changed, None
+
+
+state_strategy = {
+    "present": present_strategy,
+    "absent": absent_strategy
+}
+
+
+def core(module):
+    ensure_gitlab_package(module)
+
+    gitlab_project = module.params['project']
+    state = module.params['state']
+
+    gl = gitlab_authentication(module)
+
+    project = find_project(gl, gitlab_project)
+    # project doesn't exist
+    if not project:
+        module.fail_json(msg="project '%s' not found." % gitlab_project)
+
+    wished_badge = {
+        "link_url": module.params["link_url"],
+        "image_url": module.params["image_url"]
```

Comment:
Having a trailing comma here will make the diff smaller if another entry is added later on:
```suggestion
        "image_url": module.params["image_url"],
```